### PR TITLE
violation message locale switchable without restart eclipse

### PR DESCRIPTION
**Pull request by qxo**
This PR suitable for user who using non-english os but want to see different locale violation message

### CODE REVIEW

**Comment by romani** [comment]
@qxo, please share UI screenshots before and after update to clearly shows benefits.

**Comment by qxo** [comment]
> @qxo, please share UI screenshots before and after update to clearly shows benefits.
> 
> Please fix CI failures.



after change locale to "zh"  then do  "Check Code with Checkstyle" again, we'll Chinese violation message without restart eclipse.



for someone like me, who want to english  violation message to  compare the differences between the two languages

File: net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/builder/CheckerFactory.java [hunk 1]
```diff
@@ -247,4 +247,38 @@ private static Checker createCheckerInternal(InputSource input, PropertyResolver
...
+      checker.setLocaleLanguage(lastLocale);
+      checker.setLocaleCountry(platformLocale.getCountry());
+      LocalizedMessage.setLocale(platformLocale);
+    }
+  }
+
+  private static String getLocale() {
+    String lang = CheckstylePluginPrefs.getString(CheckstylePluginPrefs.PREF_LOCALE_LANGUAGE);
+    if (lang != null && !lang.isEmpty() && !"default".equals(lang)) {
+      lang = null;
```
Comment (Calixte) [inline net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/builder/CheckerFactory.java:280]: I'm not sure I understand what's going on here. So this method will only return null, empty string or `default`?

Comment (qxo) [reply net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/builder/CheckerFactory.java:280]: Sorry, it's refactor mistake, I'll fix it.

**Comment by qxo** [comment]
> @qxo, please share UI screenshots before and after update to clearly shows benefits.
> 
> Please fix CI failures.


The CI checkstyle-check have some issues:
* not follow the "CHECKSTYLE:OFF/CHECKSTYLE:ON" , ie : Messages.java:[177,3] ...
*  there lots of "MagicNumber/JavadocVariable" issue, why only the PR was reported!

```
[INFO] --- maven-checkstyle-plugin:3.1.1:check (checkstyle-check) @ net.sf.eclipsecs.ui ---
[INFO] There are 9 errors reported by Checkstyle 8.35 with config/checkstyle_checks.xml ruleset.
[ERROR] src/net/sf/eclipsecs/ui/preferences/CheckstylePreferencePage.java:[89,3] (javadoc) JavadocVariable: Missing a Javadoc comment.
[ERROR] src/net/sf/eclipsecs/ui/preferences/CheckstylePreferencePage.java:[182,33] (coding) MagicNumber: '3' is a magic number.
[ERROR] src/net/sf/eclipsecs/ui/preferences/CheckstylePreferencePage.java:[193,81] (coding) AvoidInlineConditionals: Avoid inline conditionals.
[ERROR] src/net/sf/eclipsecs/ui/Messages.java:[177,3] (javadoc) JavadocVariable: Missing a Javadoc comment.
[ERROR] src/net/sf/eclipsecs/ui/Messages.java:[177,3] (coding) DeclarationOrder: Variable access definition in wrong order.
[ERROR] src/net/sf/eclipsecs/ui/Messages.java:[177,24] (design) VisibilityModifier: Variable 'CheckstylePreferencePage_lblLocaleLanguage' must be private and have accessor methods.
[ERROR] src/net/sf/eclipsecs/ui/Messages.java:[179,3] (javadoc) JavadocVariable: Missing a Javadoc comment.
[ERROR] src/net/sf/eclipsecs/ui/Messages.java:[179,3] (coding) DeclarationOrder: Variable access definition in wrong order.
[ERROR] src/net/sf/eclipsecs/ui/Messages.java:[179,24] (design) VisibilityModifier: Variable 'CheckstylePreferencePage_lblLocaleLanguages' must be private and have accessor methods.
[INFO] ------------------------------------------------------------------------
```

**Comment by romani** [comment]
@qxo , please resolve violations that are introduced by your patch.

We are using special Filter that is based on git patch, so only affected code is validated, if some violations are unrelated to your changes, it might be bug in such filter, we will review them separately.

**Comment by romani** [comment]
@Bananeweizen, please put green tick of approval to be more explicit.

**Review by Bananeweizen — APPROVED** [review]

**Comment by romani** [comment]
@Calixte , this PR is ready for merge.

**Pull request merged**